sdk/data/azcosmos/example_test.go

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcosmos
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"os"
+)
+
+// This example shows you how to get started using the Azure Cosmos DB SDK for Go.
+// NewCosmosClient creates a new instance of CosmosClient with the specified values. It uses the default pipeline configuration.
+func Example() {
+
+	endpoint, _ := os.LookupEnv("SOME_ENDPOINT")
+	key, _ := os.LookupEnv("SOME_KEY")
+
+	// Create new Cosmos DB client.
+	cred, _ := NewSharedKeyCredential(key)
+	client, err := NewCosmosClient(endpoint, cred, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// All operations for Go operate on a context.Context, allowing you to control cancellation/timeout.
+	ctx := context.Background()
+
+	// ===== 1. Creating a database =====
+
+	databaseName := CosmosDatabaseProperties{Id: "databaseName"}
+	database, err := client.CreateDatabase(ctx, databaseName, nil, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// This example showcases several common operations to help you get started, such as:
+	// ===== 2. Creating a container =====
+
+	properties := CosmosContainerProperties{
+		Id: "aContainer",
+		PartitionKeyDefinition: PartitionKeyDefinition{
+			Paths: []string{"/myPartitionKey"},
+		},
+	}
+
+	throughput := NewManualThroughputProperties(400)
+
+	resp, err := database.DatabaseProperties.Database.CreateContainer(ctx, properties, throughput, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	container := resp.ContainerProperties.Container
+
+	// ===== 3. Update container properties =====
+
+	resp, err = container.Read(ctx, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	updatedProperties := CosmosContainerProperties{
+		Id: "aContainer",
+		PartitionKeyDefinition: PartitionKeyDefinition{
+			Paths: []string{"/myPartitionKey"},
+		},
+		ETag: "someEtag",
+		IndexingPolicy: &IndexingPolicy{
+			IncludedPaths: []IncludedPath{},
+			ExcludedPaths: []ExcludedPath{},
+			Automatic:     false,
+			IndexingMode:  IndexingModeNone,
+		},
+	}
+
+	// Replace container properties
+	resp, _ = container.Replace(ctx, updatedProperties, nil)
+
+	// Read the manual throughput property
+	throughputResponse, _ := container.ReadThroughput(ctx, nil)
+	_, _ = throughputResponse.ThroughputProperties.ManualThroughput()
+
+	// Replace manual throughput property
+	newScale := NewManualThroughputProperties(500)
+	_, _ = container.ReplaceThroughput(ctx, *newScale, nil)
+
+	// Migrate from manual throughput to autoscale
+	_, _ = throughputResponse.ThroughputProperties.AutoscaleMaxThroughput()
+
+	newScale = NewAutoscaleThroughputProperties(10000)
+	_, _ = container.ReplaceThroughput(ctx, *newScale, nil)
+
+	// ===== 4. Item CRUD =====
+
+	// Items in an Azure Cosmos container are uniquely identified by their id and partition key value.
+	// pk, _ := NewPartitionKey("myPartitionKey")
```

Comment:
This should be uncommented?

The value of the PK must be the value of the property, for example:

```
pk, err := NewPartitionKey("aPKValue")
if (err != nil) {
    log.Fatal(err)
}

item := map[string]string{
		"id":    "1",
		"value": "2",
                "myPartitionKey": "aPKValue"
}

itemResponse, err := container.CreateItem(ctx, pk, item, nil)
```